scripts/uosc/elements/TopBar.lua

```diff
@@ -71,7 +71,7 @@ end
 
 function TopBar:decide_enabled()
 	if options.top_bar == 'no-border' then
-		self.enabled = not state.border or state.fullscreen
+		self.enabled = not state.border or (state.title_bar ~= nil and not state.title_bar) or state.fullscreen
```

Comment:
```suggestion
		self.enabled = not state.border or not state.title_bar or state.fullscreen
```
`nil` is treated as false, no need to check for it specifically.

Reply:
No, since `nil` is treated as false, this will always make the trigger condition true.
This does not meet expectations and is not what we want.

Reply:
Why would you want to hide the top bar when it's `nil`?

Reply:
```suggestion
		self.enabled = not state.border or state.title_bar or state.fullscreen
```
That `not` in there confused me, I think this is what we want.

Reply:
```suggestion
		self.enabled = not state.border or state.title_bar == false or state.fullscreen
```
Then this or not?

Reply:
I missed it.